rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
+                                event_type = HistoryEventType.UNSTAKING
+                                event_subtype = HistoryEventSubType.STAKING_RECEIVE_ASSET
+                        elif event_type == HistoryEventType.UNKNOWN:
+                            found_unknown_event = True
+                            self.msg_aggregator.add_warning(
+                                f'Encountered unknown event type {raw_event["type"]}. Event '
+                                f'will be stored but not processed.',
+                            )
+                        fee = deserialize_asset_amount(raw_event['fee'])
+
+                        group_events.append(HistoryEvent(
+                            event_identifier=identifier,
+                            sequence_index=idx,
+                            timestamp=timestamp,
+                            location=Location.KRAKEN,
+                            location_label=self.name,
+                            amount=AssetBalance(
+                                asset=asset,
+                                balance=Balance(
+                                    amount=amount,
+                                    usd_value=ZERO,
+                                ),
+                            ),
+                            notes=None,
+                            event_type=event_type,
+                            event_subtype=event_subtype,
+                        ))
+                        if fee != ZERO:
+                            group_events.append(HistoryEvent(
+                                event_identifier=identifier,
+                                sequence_index=len(events),
+                                timestamp=timestamp,
+                                location=Location.KRAKEN,
+                                location_label=self.name,
+                                amount=AssetBalance(
+                                    asset=asset,
+                                    balance=Balance(
+                                        amount=fee,
+                                        usd_value=ZERO,
+                                    ),
+                                ),
+                                notes=None,
+                                event_type=event_type,
+                                event_subtype=HistoryEventSubType.FEE,
+                            ))
+                    except (DeserializationError, KeyError, UnknownAsset) as e:
+                        msg = str(e)
+                        if isinstance(e, KeyError):
+                            msg = f'Keyrror {msg}'
+                        self.msg_aggregator.add_error(
+                            f'Failed to read ledger event from kraken {raw_event} due to {msg}',
+                        )
+                        break
+                else:
+                    # In the case where a part of the event can't be correctly procesed change
```

Comment:
typo ... but also.. damn this is ugly


```suggestion
                    # In the case where a part of the event can't be correctly processed change
```